### PR TITLE
#40: Moved Config data model from SLC-CI-Setup to SLC-CI

### LINKED ISSUE #40: Config schema defined in downstream project is used in upstream

### PR DESCRIPTION
**Pull request by tkilias**
Fixes #40

### CODE REVIEW

File: doc/changes/changelog.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Changes
 
-* [1.2.0](changes_1.1.0.md)
+* [1.2.0](changes_1.3.0.md)
```
Comment (kaklakariada) [inline doc/changes/changelog.md:3]:
```suggestion
* [1.3.0](changes_1.3.0.md)
```

File: exasol_script_languages_container_ci/lib/config/config_data_model.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# generated by datamodel-codegen:
+#   filename:  <stdin>
```
Comment (kaklakariada) [inline exasol_script_languages_container_ci/lib/config/config_data_model.py:2]: Mark file as generated in .gitattributes

File: test/conftest.py [hunk 3]
```diff
@@ -1,13 +1,44 @@
...
+    parser.addoption(
+        DISABLE_PYDANTIC_MODEL_GENERATION, action="store_true", default=False,
+        help="Disables the generation of the pydantic models from the json schemas"
+    )
+
+
+def pytest_configure(config):
+    """
+    Some of our tests are based on the pydantic model. We need to make sure, the tests work with the newest
+    version. However, we also need regenerate the file as early as possible, before other modules import it.
```
Comment (kaklakariada) [inline test/conftest.py:32]:
```suggestion
    version. However, we also need to regenerate the file as early as possible, before other modules import it.
```

File: test/unit_tests/test_config_data_model_generator.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+from exasol_script_languages_container_ci.lib.config.data_model_generator import CONFIG_DATA_MODEL_FILE_NAME, \
+    generate_config_data_model
+
+
+def test_loading_generated_module(tmp_path):
+    config_data_model_file = tmp_path / CONFIG_DATA_MODEL_FILE_NAME
+    generate_config_data_model(config_data_model_file)
+    module = load_module(config_data_model_file)
+    assert True
```
Comment (kaklakariada) [inline test/unit_tests/test_config_data_model_generator.py:11]: Add a real assertion

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: test/unit_tests/test_config_data_model_generator.py (hunk 4 above)
Comment (tkilias) [reply test/unit_tests/test_config_data_model_generator.py:11]: the load module is more or less the actual assert, but I added a bit better assert, anyway

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**